scalding-core/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -210,6 +211,9 @@ trait TypedPipe[+T] extends Serializable {
   /** prints the current pipe to stdout */
   def debug: TypedPipe[T] = onRawSingle(_.debug)
 
+  /** adds a description to the pipe */
+  def withDescription(description: String): TypedPipe[T] = new WithDescriptionTypedPipe[T](this, description)
```

Comment:
The other PR going in now puts a common toPipe wrapper in the TypedPipe trait for setting some state on our pipes before handing them out. When that merges doing the TypedPipe extends... will be more straight forward since all typed pipe instances are just suppling asPipe